tds/src/main/java/thredds/core/ConfigCatalogInitialization.java

```diff
@@ -181,7 +181,16 @@ public synchronized boolean reread(ReadMode readMode, boolean isStartup) {
 
     switch (readMode) {
       case always:
-        if (databaseAlreadyExists) this.datasetTracker.reinit();
+        // if the database already exists, we need to close it
+        // before we reinit
+        if (databaseAlreadyExists) {
+          try {
+            this.datasetTracker.close();
+          } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
It's better to log this to `logCatalogInit` than dump the stack trace to STDERR. I think SLF4J provides a `error()` method that takes a `Throwable` as an argument.

Also, you might want to log an INFO message about  closing `datasetTracker` because the database  already exists.  Might be useful for debugging later.

Reply:
Good call - thanks!